### PR TITLE
Added Initial LDAP Auth CRUD functions and few tests

### PR DESCRIPTION
**Pull request by kbidarkar**
*) Added LDAP Authentication CRUD functions
*) Added creation and deletion tests related to LDAP Authentication
*) Added locators for LDAP Auth
*) All LDAP Information will be pulled in from the robottelo.properties
   file and the needed conf entries has been added

### CODE REVIEW

**Comment by kbidarkar** [comment]
Below are the test run results.

```
(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_create_ldap_auth_withad tests.foreman.ui.test_adusergroups
.
----------------------------------------------------------------------
Ran 1 test in 12.728s

OK

(robottelo)[kbidarka@xyz robottelo]$ nosetests -m test_delete_ldap_auth_withad tests.foreman.ui.test_adusergroups
.
----------------------------------------------------------------------
Ran 1 test in 14.119s

OK
```

File: robottelo/ui/ldapusergroups.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        self.wait_for_ajax()
+        strategy1, value1 = locators["ldapserver.ldap_delete"]
+        element = self.wait_until_element((strategy1, value1 % name))
+        if element is None:
+            raise UINoSuchElementError(
+                u'Could not select the entity "{0}" for deletion.'
+                .format(name)
+            )
+        element.click()
+        self.handle_alert(really)
```
Comment (Ichimonji10) [inline robottelo/ui/ldapusergroups.py]: This file can be refactored a bit and dropped down to 80 lines.

```diff
diff --git a/robottelo/ui/ldapusergroups.py b/robottelo/ui/ldapusergroups.py
index 11b6dae..2005fa8 100644
--- a/robottelo/ui/ldapusergroups.py
+++ b/robottelo/ui/ldapusergroups.py
@@ -15,66 +15,54 @@ class LdapAuth(Base):
                account_passwd=None, account_basedn=None,
                account_grpbasedn=None, ldap_filter=False, otf_register=True):
         """Create new ldap auth source from UI."""
-        if self.wait_until_element(locators["ldapsource.new"]):
-            self.wait_until_element(locators["ldapsource.new"]).click()
-            if self.wait_until_element(locators["ldapserver.name"]):
-                self.field_update("ldapserver.name", name)
-                self.field_update("ldapserver.server", server)
-                self.wait_until_element(locators["ldapserver.ldaps"]).click()
-                if port:
-                    self.field_update("ldapserver.port", port)
-                Select(
-                    self.find_element(locators["ldapserver.server_type"])
-                ).select_by_visible_text(server_type)
-            self.wait_until_element(
-                tab_locators["ldapserver.tab_account"]
-            ).click()
-            if self.wait_until_element(
-                locators["ldapserver.acc_user"]
-            ) is None:
-                raise UINoSuchElementError(
-                    u'Could not select the attributes Tab.'
-                )
-            self.field_update("ldapserver.acc_user", account_user)
-            self.field_update("ldapserver.acc_passwd", account_passwd)
-            self.field_update("ldapserver.basedn", account_basedn)
-            self.field_update("ldapserver.group_basedn", account_grpbasedn)
-            if ldap_filter:
-                self.wait_until_element(
-                    locators["ldapserver.ldap_filter"]
-                ).click()
-            self.wait_until_element(
-                locators["ldapserver.otf_register"],
-                otf_register
-            ).click()
-            self.wait_until_element(
-                tab_locators["ldapserver.tab_attributes"]
-            ).click()
-            if self.wait_until_element(
-                locators["ldapserver.loginname"]
-            ) is None:
-                raise UINoSuchElementError(
-                    u'Could not select the account Tab.'
-                )
-            self.field_update("ldapserver.loginname", login_name)
-            self.field_update("ldapserver.firstname", first_name)
-            self.field_update("ldapserver.surname", surname)
-            self.field_update("ldapserver.mail", mail)
-            if photo:
-                self.field_update("ldapserver.photo", photo)
-            self.wait_until_element(common_locators["submit"]).click()
-            self.wait_for_ajax()
+        if not self.wait_until_element(locators['ldapsource.new']):
+            return
+        self.wait_until_element(locators['ldapsource.new']).click()
+        if self.wait_until_element(locators['ldapserver.name']):
+            self.field_update('ldapserver.name', name)
+            self.field_update('ldapserver.server', server)
+            self.wait_until_element(locators['ldapserver.ldaps']).click()
+            if port:
+                self.field_update('ldapserver.port', port)
+            Select(
+                self.find_element(locators['ldapserver.server_type'])
+            ).select_by_visible_text(server_type)
+        self.wait_until_element(tab_locators['ldapserver.tab_account']).click()
+        if self.wait_until_element(locators['ldapserver.acc_user']) is None:
+            raise UINoSuchElementError(u'Could not select the attributes Tab.')
+        self.field_update('ldapserver.acc_user', account_user)
+        self.field_update('ldapserver.acc_passwd', account_passwd)
+        self.field_update('ldapserver.basedn', account_basedn)
+        self.field_update('ldapserver.group_basedn', account_grpbasedn)
+        if ldap_filter:
+            self.wait_until_element(locators['ldapserver.ldap_filter']).click()
+        self.wait_until_element(
+            locators['ldapserver.otf_register'],
+            otf_register
+        ).click()
+        self.wait_until_element(
+            tab_locators['ldapserver.tab_attributes']
+        ).click()
+        if self.wait_until_element(locators['ldapserver.loginname']) is None:
+            raise UINoSuchElementError(u'Could not select the account Tab.')
+        self.field_update('ldapserver.loginname', login_name)
+        self.field_update('ldapserver.firstname', first_name)
+        self.field_update('ldapserver.surname', surname)
+        self.field_update('ldapserver.mail', mail)
+        if photo:
+            self.field_update('ldapserver.photo', photo)
+        self.wait_until_element(common_locators['submit']).click()
+        self.wait_for_ajax()
 
     def search(self, name, really=False):
         """Searches existing ldap auth source from UI."""
         Navigator(self.browser).go_to_ldap_auth()
         self.wait_for_ajax()
-        strategy1, value1 = locators["ldapserver.ldap_servername"]
+        strategy1, value1 = locators['ldapserver.ldap_servername']
         element = self.wait_until_element((strategy1, value1 % name))
         if element is None:
             raise UINoSuchElementError(
-                u'Could not search the entity "{0}".'
-                .format(name)
+                u'Could not search the entity "{0}".'.format(name)
             )
         return element
 
@@ -82,12 +70,11 @@ class LdapAuth(Base):
         """Deletes existing ldap auth source from UI."""
         Navigator(self.browser).go_to_ldap_auth()
         self.wait_for_ajax()
-        strategy1, value1 = locators["ldapserver.ldap_delete"]
+        strategy1, value1 = locators['ldapserver.ldap_delete']
         element = self.wait_until_element((strategy1, value1 % name))
         if element is None:
             raise UINoSuchElementError(
-                u'Could not select the entity "{0}" for deletion.'
-                .format(name)
+                u'Could not select the entity "{0}" for deletion.'.format(name)
             )
         element.click()
         self.handle_alert(really)
```

File: tests/foreman/ui/test_adusergroups.py [hunk 2]
```diff
@@ -19,12 +25,21 @@ def test_create_ldap_auth_withad(self):
...
+        with Session(self.browser) as session:
+            make_ldapauth(session, name=server_name, server=self.ldap_hostname,
+                          server_type=LDAP_SERVER_TYPE['ad'],
+                          login_name=LDAP_ATTR['login_ad'],
+                          first_name=LDAP_ATTR['firstname'],
+                          surname=LDAP_ATTR['surname'], mail=LDAP_ATTR['mail'],
+                          account_user=self.ldap_username,
+                          account_passwd=self.ldap_passwd,
+                          account_basedn=self.ldap_basedn,
+                          account_grpbasedn=self.ldap_grpbasedn)
```
Comment (Ichimonji10) [inline tests/foreman/ui/test_adusergroups.py]: Can you just format this a bit differently?

```python
        with Session(self.browser) as session:
            make_ldapauth(
                session,
                name=server_name,
                server=self.ldap_hostname,
                server_type=LDAP_SERVER_TYPE['ad'],
                login_name=LDAP_ATTR['login_ad'],
                first_name=LDAP_ATTR['firstname'],
                surname=LDAP_ATTR['surname'],
                mail=LDAP_ATTR['mail'],
                account_user=self.ldap_username,
                account_passwd=self.ldap_passwd,
                account_basedn=self.ldap_basedn,
                account_grpbasedn=self.ldap_grpbasedn,
            )
```

I find it much easier to browse through the arguments in this form.

File: tests/foreman/ui/test_adusergroups.py [hunk 3]
```diff
@@ -36,9 +51,19 @@ def test_delete_ldap_auth_withad(self):
...
```
Comment (Ichimonji10) [inline tests/foreman/ui/test_adusergroups.py]: Same request.
```python
        with Session(self.browser) as session:
            make_ldapauth(
                session,
                name=server_name,
                server=self.ldap_hostname,
                server_type=LDAP_SERVER_TYPE['ad'],
                login_name=LDAP_ATTR['login_ad'],
                first_name=LDAP_ATTR['firstname'],
                surname=LDAP_ATTR['surname'],
                mail=LDAP_ATTR['mail'],
                account_user=self.ldap_username,
                account_passwd=self.ldap_passwd,
                account_basedn=self.ldap_basedn,
                account_grpbasedn=self.ldap_grpbasedn,
            )
```

File: docs/api/robottelo.ui.rst [hunk 4]
```diff
@@ -73,6 +73,11 @@
 
 .. automodule:: robottelo.ui.hosts
 
+:mod:`robottelo.ui.ldapusergroups`
+----------------------------------
+
+.. automodule:: robottelo.ui.ldapusergroups
```
Comment (Ichimonji10) [inline docs/api/robottelo.ui.rst]: :+1:

**Comment by Ichimonji10** [comment]
ACK pending comments.

**Comment by sthirugn** [comment]
ACK

**Pull request merged**